tests/publiccloud/aistack_basic.pm

```diff
@@ -0,0 +1,89 @@
+use Mojo::Base 'publiccloud::basetest';
+use testapi;
+use serial_terminal 'select_serial_terminal';
+use publiccloud::utils qw(is_byos registercloudguest register_openstack);
+use publiccloud::ssh_interactive 'select_host_console';
+use strict;
+use warnings;
+use utils;
+use publiccloud::utils;
+use File::Basename;
+
+sub run {
+    my ($self, $args) = @_;
+    #Download unpublished DMS package and install rpm
+    my $ins_rke2 = get_var('RKE2_URL');
+    my $ins_kubectl = get_var('KUBECTL_URL');
+    my $ins_helm = get_var('HELM_URL');
+    #my $add_repo = get_var('HELM_REPO');
+   
+
+    #Create aws instance
+    my $provider;
+    my $instance;
+    select_host_console();
+
+    my $aws_test = get_var('PUBLIC_CLOUD_AWS_TEST', 'basic_sanity');
+    if ($aws_test) {
+        $instance = $self->{my_instance} = $args->{my_instance};
+        $provider = $self->{provider} = $args->{my_provider};    # required for cleanup
+    } else {
+        $provider = $self->{provider} = $self->provider_factory();    # required for cleanup
+        $instance = $self->{my_instance} = $provider->create_instance(check_guestregister => is_openstack ? 0 : 1);
+    }
+
+    registercloudguest($instance) if is_byos();
```

Comment:
Likewise in same place in run_ltp, by experienced old fault, we need to add the same second condition too (see !$qam) :slightly_smiling_face:.

Reply:
Fixed